doc/vim-lsp.txt

```diff
@@ -505,6 +505,21 @@ g:lsp_hover_conceal                                   *g:lsp_hover_conceal*
 
     To override this setting per server, see |vim-lsp-hover_conceal|.
 
+g:lsp_location_goto_fn				*g:lsp_hover_conceal*
+    Type: |String|
+    Default: `""`
+
+    Name of the customizable function to jump to a location (e.g. in use of
+    |LspDefinition|, |LspImplementation|, etc.).
+
+    Example:
+	function! LspGoTo(filename)
```

Comment:
Can we use dict instead of just `filename` this would allow us to add other information and be consistent with others such as line number, col and so on.

Reply:
Surely we can, but could you be specific on what to put in the dict?
The reason that I didn't pass line number and col is because they were already handled by the line
`execute l:cmd . ' | call cursor('.l:loc['lnum'].','.l:loc['col'].')'`.

p.s. I don't know why CI failed. Didn't see any useful information to fix the failed CI.

Reply:
Can you guide me on what I should modify for this to be merged?

Reply:
instead of `function! LspGoTo(filename)` it should be something like `function! LspGoto(options)` where `options` is a dict.

```vim
function! LspGoTo(options)
    let l:cmd = 'tabedit '.a:options['filename']
    execute l:cmd
endfunction
```

I should also be able to customize different keybindings for goto function. currently this only does for one.

Another option would be to support args in `LspDefinition` so anyone can customize it.

```
nnoremap <buffer> <C-]> :<C-u>LspDefinition<CR>
nnoremap <buffer> <C-t> :<C-u>LspDefinition --callback=function('s:custom_go_to')<CR>
```